docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+* `TrippieExceedBudgetException` is thrown when expenses in the saved trip exceeds the budget given by user.
+
+
+## 4.0 Implementation [DG to complete by Wed 28 Oct (soft)]
+
+This section elaborates on some unique details about how certain features are implemented.
+
+### 4.1 Multiple Trips Structure (Ivander)
+
+Trippie is built for travelers, those who most probably has more than a single trip in mind. Therefore, the multiple trips feature is designed so that someone can easily work and switch between many trips.
+
+All your `Trip` objects are stored in a java ArrayList object called tripList inside the `TrippieData` object. In this `TrippieData` object, an attribute called `currentTrip` is used to point to `Trip` object inside the ArrayList that is currently being worked on.
+
+Multiple trips feature is designed to **minimize program startup time** and **minimize memory usage**. These aims are achieved by:
+1. Reading only one Master File `trippie.txt` that will be loaded on startup.
+2. Saving each trip in separate files.
+3. Loading only the user's most recently edited trip on startup. This is set as the *default trip* which will be set as the *current trip* during runtime.
+4. It is only possible to work on one `Trip` at a single point in time.
+5. The commands `new trip` and `load trip` will change the *current trip*.
+6. Current trips are automatically saved after each command execution.
+
+The format of `trippie.txt` is as follows:
+```
+DEFAULT <trip_index>
+<trip_index - 1>,<trip_name>,<trip_date (dd-mm-yyyy)>,<trip_max_day>
+1,Amazing Bali,03-04-2021,3
+2,Australia,01-01-2021,8
+3,Wonderful Singapore,02-01-2021,9
+```
+Here the `trip_name` serves both as the Trip's name and the file name, hence it should not contain invalid characters for file names, i.e. `<>:"/\|?*.` A sample directory structure containing 3 trips is as the following:
+```
+trippie_data
+    ├── Amazing Bali.txt
+    ├── Australia.txt
+    ├── Wonderful Singapore.txt
+    └── trippie.txt
+```
+
+
+
+### 4.2 Sorting Place List (Kian En)
+
+The way Trippie sorts its place list, regardless of the order the places are added by the user, is through **bubble sort**. The sorting algorithm is called every time a new place is added by the user. 
+
+When listing either places, Trippie will sort them based on both *Day* and *Time*. Hence, there is a need to use a **stable** sorting algorithm. Bubble sort is used for its simplicity and easy implementation. Furthermore, since this sorting algorithm is called every time a new place is added, the list will always be sorted, allowing the time complexity of the sorting algorithm to be **O(n)**.
+
+Likewise when sorting expenses, Trippie will sort them based on *Day*.
+
+### 4.3 Budget and Expenses (Shawn)
+
+Trippie provides travellers an easy and convenient way to track expenses while constantly ensuring that the budget is not exceeded.
+
+<kbd>
+
+![Expenses UML Diagram](https://i.imgur.com/xuZNMxG.jpg)
```

Comment:
Perhaps void could be used to signify methods without return type.
![image](https://user-images.githubusercontent.com/49357313/97660630-25286200-1aad-11eb-8206-a885954abce4.png)